### PR TITLE
Add Changelog Entries for 1.12 release.

### PR DESCRIPTION
**Pull request by orsenthil**
## High-level description

* Release Notes Proposal for DC/OS. https://jira.mesosphere.com/browse/QUALITY-1879 

### Description

This is our desired format for the change log. This gives guidelines on how to add change log entries and adds entries to already committed changes in 1.12. 

As new features are added, reviewers could request developers to add an entry to this file following the suggested format. 

Note - The format, style and the methodology is work in progress during early 1.12 timeframe.  We will evolve it as realize the improvements that can be made to this process.

### CODE REVIEW

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
@mesosphere-mergebot label ready for review

File: CHANGES.md [hunk 1]
```diff
@@ -1 +1,48 @@
...
+* Update Metronome to 0.5.0 (DCOS_OSS-2338)
+
+* Release cosmos v0.6.0. (DCOS_OSS-2195)
+
+* DC/OS API endpoint to determine if the installation is Open or Enterprise. (DCOS_OSS-2283)
+
+
+### Security Updates
+
+* Bump curl from 7.48 to 7.59 (DCOS-21557)
```
Comment (vespian) [inline CHANGES.md]: I am sorry if this question has already been asked somewhere, but is there a point in linking private Jiras in Open DC/OS PR?

Comment (orsenthil) [reply CHANGES.md]: Yeah, I should have an Open DC/OS  JIRA. @ahairape  do we have one for this or shall I create it?

**Comment by jgehrcke** [comment]
Woop, again some process. Thanks, Senthil! 

Is it okay if in addition to a fine-grained review I also add a corresponding commit on top?

Things I have noticed there so far:

- One inconsistency is between `External Projects like Mesos and Marathon shall provide a link to their published changelogs.`  and `### Mesos Changelog`. I think we agreed before that we do not want to carry the Mesos changelog and the Marathon changelog here. Not sure how to resolve that given that we want to have 'additions only' for individual 'standard process PRs'. I think that means that Marathon/Mesos bump PRs are allowed to not have a changelog entry (because at PR submission time it is not yet known which version will end up in a public DC/OS release). And that the correct changelog entry must be entered via an 'editorial changelog PR' (which as opposed to 'standard process PRs' can bypass the standard process).

- https://jira.mesosphere.com/browse/DCOS_OSS-2162 is actually a security fix

- There are some grammar inconsistencies that I'd like to take care of.

**Comment by orsenthil** [comment]
> I think we agreed before that we do not want to carry the Mesos changelog and the Marathon changelog here.

Yeah. I thought, we might need a placeholder to add link to the Mesos / Marathon changelog. An indicator that , DC/OS includes Mesos x.y+ and Marathon x.y+ at this moment easily recognizable when reading the changelog.

File: CHANGES.md [hunk 2]
```diff
@@ -1 +1,48 @@
...
+
+* Entry with no-newlines. (DCOS_OSS_JIRA)
+<new-line>
+* Entry two with no-newlines. (DCOS_OSS_JIRA_2)
+```
+
+
+### Fixed and improved
+
+* Windows-based pkgpanda builds (DCOS_OSS-1899)
```
Comment (jgehrcke) [inline CHANGES.md]: Enabled Windows-based pkgpanda builds. (DCOS_OSS-1899)

File: CHANGES.md [hunk 3]
```diff
@@ -1 +1,48 @@
...
+
+* Upgrade REX-Ray to v0.11.2. (DCOS_OSS-2245)
```
Comment (jgehrcke) [inline CHANGES.md]: - Can this go into a 'Notable changes' section?

- Upgraded to ...

File: CHANGES.md [hunk 4]
```diff
@@ -1 +1,48 @@
...
+
+* Moves the dcos-metrics Prometheus producer from port 9273 to port 61091. Using a port which is not available to Mesos (any unoccupied port > 61000) avoids the need to update anything. (DCOS-21594)
```
Comment (jgehrcke) [inline CHANGES.md]: The issue number leads me to an issue which seems unrelated to the text.

Comment (jgehrcke) [reply CHANGES.md]: (I'd like to work on the text, but dunno which issue to look up.)

Comment (jgehrcke) [reply CHANGES.md]: - DC/OS Metrics: moved the ...

File: CHANGES.md [hunk 5]
```diff
@@ -1 +1,48 @@
...
+
+* Update Metronome to 0.5.0 (DCOS_OSS-2338)
```
Comment (jgehrcke) [inline CHANGES.md]: - I think this should go into a 'Notable changes' section.

- Full stop missing.

Comment (jgehrcke) [reply CHANGES.md]: Format should be "Updated Metronome to ..."

File: CHANGES.md [hunk 6]
```diff
@@ -1 +1,48 @@
...
+
+* Release cosmos v0.6.0. (DCOS_OSS-2195)
```
Comment (jgehrcke) [inline CHANGES.md]: I think for Cosmos we should have the individual relevant changes here.

File: CHANGES.md [hunk 7]
```diff
@@ -1 +1,48 @@
...
+
+* DC/OS API endpoint to determine if the installation is Open or Enterprise. (DCOS_OSS-2283)
```
Comment (jgehrcke) [inline CHANGES.md]: Added a DC/OS API endpoint to distinguish the 'open' and 'enterprise' build variants. (DCOS_OSS-2283)

File: CHANGES.md (hunk 1 above)
Comment (jgehrcke) [reply CHANGES.md]: Updated cURL to 7.59. (xxx)

File: CHANGES.md [hunk 8]
```diff
@@ -1 +1,48 @@
...
+
+
+### Security Updates
+
+* Bump curl from 7.48 to 7.59 (DCOS-21557)
+
+* Update openssl to 1.0.2n (DCOS_OSS-1903)
```
Comment (jgehrcke) [inline CHANGES.md]: Updated OpenSSL to 1.0.2n. (DCOS_OSS-1903)

File: CHANGES.md (hunk 1 above)
Comment (orsenthil) [reply CHANGES.md]: Addressed.

**Comment by orsenthil** [comment]
Thanks for the comments @jgehrcke  and @vespian. 

I have addressed your review comments. This is shaping well.

File: CHANGES.md (hunk 3 above)
Comment (orsenthil) [reply CHANGES.md]: We should give guidance on what qualifies as Notable change.

I don't know why this qualifies as notable change. Could you please share why this should be mentioned in "Notable changes" section.  I am assuming you already thought about this.

File: CHANGES.md (hunk 4 above)
Comment (orsenthil) [reply CHANGES.md]: Addressed this comment.  Included the correct JIRA ID.

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

File: CHANGES.md (hunk 3 above)
Comment (jgehrcke) [reply CHANGES.md]: Great point. My rough thinking here is that if the content of the item does not directly communicate what has been "improved" or "fixed" then it's not suitable for the "Fixed and improved" section, but should go into a more neutral section.

Or, in other words, for every point in "Fixed and improved" I want to know what about the change has fixed or improved something, and it's not really up for interpretation.

These items should mainly be bugfixes, performance improvements, resilience improvements, UX improvements, etc.

The following are examples for notable changes (changes of significance, but the kind of the significance is not communicated, and depends on the point of view):
"Platform: the Java Developer Kit was upgraded to JDK 8u152."
"DC/OS is now compatible with Docker version 17.05.0"
"DC/OS is now compatible with RHEL 7.4."

This is brainstorm. Can surely be reduced to something more concise. Taking note of this for a 'guidelines' document.

File: CHANGES.md [hunk 9]
```diff
@@ -1 +1,42 @@
...
+* Added a DC/OS API endpoint to distinguish the 'open' and 'enterprise' build variants. (DCOS_OSS-2283)
+
+
+### Security Updates
+
+* Update cURL to 7.59. (DCOS_OSS-2367)
+
+* Updated OpenSSL to 1.0.2n. (DCOS_OSS-1903)
+
+* Mesos modules enables file based ZK configuration (DCOS_OSS-2162)
```
Comment (jgehrcke) [inline CHANGES.md]: Mesos does not expose ZooKeeper credentials anymore via its state JSON document. (DCOS_OSS-2162)

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

File: CHANGES.md [hunk 10]
```diff
@@ -1 +1,42 @@
-## DC/OS 1.12.0
\ No newline at end of file
+## DC/OS 1.12.0
+
+```
+* For any significant improvement to DC/OS add an entry to Fixed and Improved section.
```
Comment (margaret) [inline CHANGES.md]: How are we defining `significant`?

Comment (jgehrcke) [reply CHANGES.md]: We need to work on the guidelines. The idea is that all user-facing changes must have a corresponding changelog entry.

The idea is each individual pull request needs to justify why it should _not_ need to come with a changelog entry (it must then explain why the change is purely internal / not user-facing), instead of the other way round.

Comment (orsenthil) [reply CHANGES.md]: We have to evolve the guidelines. We have to socialize the idea, understand the kind of situations we land into with respect to the addition of changelog entries, and then we could add this to the PR template.

Comment (orsenthil) [reply CHANGES.md]: > How are we defining significant?

Our thinking is, features, bug fixes, bumps are significant.The final goal is about communicating what the customer can expect/see from this release. 

Things like fixing a test, refactors, documentation improvements, addition of tests _need not_ be included in the change-log. 

I was not thinking of explaining this,  I was thinking that it is understandable by the PR creator. 
The challenge I see is enforcing the process of requesting for change log when reviewing.Let's see how it goes.

**Comment by orsenthil** [comment]
EE side didn't build properly because Github didnt see the correct SHA. (This can happen with the open side was force pushed, which I did in addressing a review comments of this PR).

Since the PR is already approved by two people, I am applying the ship-it label.

**Comment by orsenthil** [comment]
@mesosphere-mergebot label ship-it
